sos/plugins/azure.py

```diff
@@ -33,10 +33,11 @@ def setup(self):
             "/etc/default/kv-kvp-daemon-init",
             "/etc/waagent.conf",
             "/sys/module/hv_netvsc/parameters/ring_size",
-            "/sys/module/hv_storvsc/parameters/storvsc_ringbuffer_size",
-            "/var/log/azure*"
+            "/sys/module/hv_storvsc/parameters/storvsc_ringbuffer_size"
         ])
 
+        self.add_copy_spec("/var/log/azure*", sizelimit=10)
```

Comment:
This will not work if `/var/log/azure*` expands to a directory (`/var/log/azure/`). You need to use a glob pattern that will match a list of log files (e.g. `*.log` is common) to have the size limit effectively applied. There's more information on this and the other global log limiting options in the GitHub wiki.

Reply:
The `/var/log/azure*` glob does expand to a directory that will contain other directories and/or files. The Azure extensions provided perform their own logging under this directory, and don't always follow a common filename pattern.

I think deleting this PR is probably the best option here, but let me know your thoughts.

Reply:
We can still limit them (assuming they are in a flat directory structure - i.e. no more subdirectories beneath `/var/log/azure`) - the glob just needs to use `/var/log/azure/*` in that case (all the files within the directory).

The lack of any pattern or structure to the names does mean that we cannot properly implement `--all-logs` for this plugin, since there is no way for us to tell "current" logs apart: the option is mainly used with traditional UNIX-style log rotation, where "old" logs have an appended counter or date string.

What will happen today in the event that the limit is reached is that we will collect logs in most recently written order up until the limit is reached: this is still useful in many cases since the most recently written data is often of most interest, so even though we can't implement everything for Azure I still think it's worth going ahead and adding this.

Reply:
Thanks for that explanation! That explains why I wasn't seeing the `sizelimit` working when testing the plugin.

The /var/log/azure directory will not be a flat structure, unfortunately. If it is helpful, I can create a test virtual machine and install some plugins to show you what I'm working with.

Reply:
Thanks - that'd definitely be useful. The other (more complex, but sometimes more effective) approach is to see if there is something in the Azure environment that can be interrogated to get a list of log files, and that we can use to prioritise or select them for collection. It's uncommon today for plain logs but we do have a couple of other bits of data collection that work this way (e.g. parse a list of things to collect from some command output, and then generate a list of files to collect from that).

Reply:
Side note: we're just setting up a meeting with some of the core folks to address outstanding issues in the content size limit handling in sos. Some of this may land in time for the next release - the main work at the moment is introducing size limiting for command output, and making sure that command and file size limits behave consistently.

As part of this work I'd like to see if we can tackle some of the long-standing limitations in the general `add_copy_spec()` size limiting: some of those changes may help with this particular use case.

Reply:
Sorry, just now getting time to come back to this. Below is an example of the Azure Recovery Services extension logging. The directory depth is indeterminate, and log files don't follow normal filename conventions (e.g., `*.log`).

~~~
$ sudo find /var/log/azure -type f
/var/log/azure/Microsoft.Azure.RecoveryServices.SiteRecovery.Linux/1.0.0.9100/SiteRecoveryExtension
/var/log/azure/Microsoft.Azure.RecoveryServices.SiteRecovery.LinuxRHEL7/1.0.0.9100/SiteRecoveryExtension
~~~

As a temporary workaround, would it be possible to perform a `find /var/log/azure -type f` as shown above, then loop through all the results and add them individually via `add_copy_spec()`?

The work you talk about with `add_copy_spec()` and size limiting does sound promising, so if you think it's worth waiting on that, I certainly can.